### PR TITLE
[FIX] Orphaned change file errors are not surfaced correctly in InfoCenter

### LINKED ISSUE #4944: BUG - Orphaned change file errors are not surfaced correctly in InfoCenter
### Description

When an SAP Fiori Adaptation Project references a fragment or controller extension file that no longer exists (an orphaned change), UI5 logs an error to the browser console.

The current handling has two issues:
1. The InfoCenter message does not reflect the actual browser error. Instead, it displays a fixed localized template:
   `The "{filePath}" file referenced by the "{changeFileName}" change was not found. To resolve this error, delete the change file`
   This hides important debugging information such as the failing URL and the underlying root cause.
2. The InfoCenter title is too specific. The current title, `"Missing File Detected"`, assumes a particular root cause, while the same error interception path can also be triggered by other flex-apply failures, such as the use of deprecated controls.

### Steps to Reproduce
1. Open an SAP Fiori Adaptation Project.
2. Create or keep a change that references a fragment or controller extension file.
3. Delete the referenced file without removing the corresponding change.
4. Reload or preview the application.
5. Observe the browser console error and the InfoCenter notification.

### Expected results
- The InfoCenter displays the actual error message reported by the browser/UI5 runtime.
- Relevant debugging details (such as the failing URL and root cause) are preserved and visible to the user.
- The InfoCenter title remains generic and applicable to different flex-apply failure scenarios.

### Actual results
- The browser console contains the real UI5 error with detailed diagnostic information.
- The InfoCenter replaces the original error with the fixed message:
  `The "{filePath}" file referenced by the "{changeFileName}" change was not found. To resolve this error, delete the change file`
- Important troubleshooting information is hidden from the user.
- The title `"Missing File Detected"` suggests a specific root cause even when the failure was caused by a different flex-apply issue (for example, usage of deprecated controls).

#### Version/Components/Environment	
Add any other context about the problem here
OS:                      
- [x] Mac OS
- [x] Windows
- [ ] Other

### Problem
{describe the problem} 

### Fix
{describe the fix} 

### Why was it missed
{Some explanation why this issue might have been missed during normal development/testing cycle}  

### How can we avoid this
{if we don’t want to see this type of issues anymore what we should do to prevent}

### PR DESCRIPTION
**Pull request by IvoSG**
## Description

Fixes orphaned change file errors not being surfaced correctly in the InfoCenter for SAP Fiori Adaptation Projects. Previously, when UI5 failed to load a fragment or controller extension file referenced by a change, the InfoCenter displayed a fixed localized message (e.g., `The "{filePath}" file referenced by the "{changeFileName}" change was not found...`) instead of the actual browser/UI5 error. This hid important debugging details like the failing URL and root cause.

### Key Changes

- **`change-file-validator.ts`**: Refactored the orphaned change detection logic:
  - Replaced the `Map<string, OrphanedChangeEntry>` (which stored change metadata for composing a fixed message) with a simpler `Set<string>` of module name patterns.
  - The error handler now forwards the **original browser error message** (with UI5 log timestamp stripped) directly to the InfoCenter instead of a templated localized string.
  - Changed `initOrphanedChangeDetection` from `async` to synchronous — it now wraps `console.error` **immediately** on call, buffering up to 200 early errors before the changes list is fetched. Once loaded, buffered messages are replayed against the known module names.
  - Returns a `cancel()` function to allow external callers to stop detection and restore `console.error`.
  - The safety timeout now calls `cancel()` instead of just `restore()`, ensuring the timeout handle is also cleared.

- **`init.ts`**: Updated to use the new synchronous API — calls `initOrphanedChangeDetection()` at the start of `init` and uses the returned `cancel` function to stop detection when the UI5 version check fails early.

- **`messagebundle.properties`**: Replaced `ADP_ORPHANED_CHANGE_ERROR_TITLE` ("Missing File Detected") and `ADP_ORPHANED_FILE_DESCRIPTION` with a single, more generic `ADP_CHANGE_ERROR_TITLE` ("Change Error").

- **Tests**: Updated and extended with new test cases covering buffered replay, cancel behavior, buffer cap (200 entries), timestamp stripping, and empty `moduleName` edge cases.

## Type of change
- [x] Bug (non-breaking change which fixes an issue)

## How have you tested?
Unit tests updated and extended to cover all new and modified behaviors, including buffered replay, cancel, and timestamp stripping.

## Checklist:
- [x] The code conforms to the [general development principles](https://github.com/SAP/open-ux-tools/blob/main/docs/Guidelines.md)
- [x] Supplied as many details as possible on this change
- [x] The code is easy to read and maintainable by others
- [ ] Corresponding changes to the documentation has been done
- [x] Already existing and new unit tests pass locally

Resolves https://github.com/SAP/open-ux-tools/issues/4944

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.27.6`

- Event Trigger: `issue_comment.edited`
- Correlation ID: `26263620-8030-11f1-8436-077c33f1a8e3`
</details>

### CODE REVIEW

**Review by lfindlaysap — DISMISSED** [review]

File: packages/preview-middleware-client/src/adp/change-file-validator.ts [hunk 1]
```diff
@@ -112,47 +108,101 @@ function createErrorHandler(
...
 
-    const handler = createErrorHandler(moduleNameMap, restore);
+    const cancel = (): void => {
+        cancelled = true;
+        restore();
+        clearTimeout(safetyTimeout);
+    };
 
-    const safetyTimeout = setTimeout(restore, 60_000);
+    const safetyTimeout = setTimeout(cancel, 60_000);
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/src/adp/change-file-validator.ts:141]: The forward-reference is safe by construction (JavaScript const closures capture by binding, and cancel is never called before safetyTimeout is assigned). Restructuring would add noise for no correctness gain.

File: packages/preview-middleware-client/src/adp/change-file-validator.ts [hunk 2]
```diff
@@ -112,47 +108,101 @@ function createErrorHandler(
...
+            handler = createErrorHandler(moduleNames, cancel);
+
+            // Replay any errors emitted before the changes list was loaded.
+            for (const message of buffer) {
+                handler(message);
+                if (moduleNames.size === 0) {
+                    break;
+                }
+            }
+            buffer.length = 0;
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/src/adp/change-file-validator.ts:201]: No change needed

File: packages/preview-middleware-client/src/adp/change-file-validator.ts [hunk 3]
```diff
@@ -50,58 +42,62 @@ function isFragmentOrCodeExtChange(change: Change): change is RelevantChange {
...
-        const path = change.changeType === CHANGE_TYPE.addXML ? change.content?.fragmentPath ?? '' : change.content?.codeRef ?? '';
-        const changeFileName = `${change.fileName}.${change.fileType ?? 'change'}`;
-        const key = change.moduleName ?? `${prefix}/changes/${path}`;
-
-        map.set(key, { changeFileName, filePath: path, changeType: change.changeType });
+        const path =
+            change.changeType === CHANGE_TYPE.addXML
+                ? change.content?.fragmentPath ?? ''
+                : change.content?.codeRef ?? '';
+        moduleNames.add(change.moduleName || `${prefix}/changes/${path}`);
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/src/adp/change-file-validator.ts]: Fixed in [80ba52](https://github.com/SAP/open-ux-tools/pull/4945/changes/80ecbdf461916223fab3e1db680835bd2080ba52)

File: packages/preview-middleware-client/src/adp/change-file-validator.ts [hunk 4]
```diff
@@ -112,47 +108,101 @@ function createErrorHandler(
...
     const restore = (): void => {
         consoleRef.error = originalConsoleError;
     };
 
-    const handler = createErrorHandler(moduleNameMap, restore);
+    const cancel = (): void => {
+        cancelled = true;
+        restore();
+        clearTimeout(safetyTimeout);
+    };
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/src/adp/change-file-validator.ts:139]: Fixed in [80ba52](https://github.com/SAP/open-ux-tools/pull/4945/changes/80ecbdf461916223fab3e1db680835bd2080ba52)

File: packages/preview-middleware-client/test/unit/adp/change-file-validator.test.ts [hunk 5]
```diff
@@ -144,16 +156,15 @@ describe('change-file-validator', () => {
 
         await initOrphanedChangeDetection();
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/test/unit/adp/change-file-validator.test.ts:157]: Fixed in [80ba52](https://github.com/SAP/open-ux-tools/pull/4945/changes/80ecbdf461916223fab3e1db680835bd2080ba52)

File: packages/preview-middleware-client/test/unit/adp/change-file-validator.test.ts [hunk 6]
```diff
@@ -97,15 +107,17 @@ describe('change-file-validator', () => {
             }
         });
 
-        await initOrphanedChangeDetection();
+        initOrphanedChangeDetection();
```
[the remark this answers is not in the document]
Comment (IvoSG) [reply packages/preview-middleware-client/test/unit/adp/change-file-validator.test.ts:110]: Fixed in [80ba52](https://github.com/SAP/open-ux-tools/pull/4945/changes/80ecbdf461916223fab3e1db680835bd2080ba52)

**Review by lfindlaysap — APPROVED** [review]

**Review by nikmace — APPROVED** [review]
Changes look good
Changeset is OK
Coverage is good
Did not test manually
